### ISSUE TITLE
How to measure the "relatively stable" milestone?

### ISSUE THREAD

**Issue by jmfederico** [el1t/uBlock-Safari#34]
Hi,

Is there a path to know when this plugin will be considered "relatively stable"?

If not, should there be one? I've seen projects stay forever in "pre-release" phase, and would like this one not to be one.

**Comment by el1t** [comment]
Good point, and I think releasing it to the extension gallery would make both installation and updates much easier. Of course it'd be nice to have as much feature parity with the Chrome version as possible. My primary concern for release, though, are unblocked connections. (For example, when  Safari allows a connection that Chrome would have blocked.)

I'm open to suggestions—is there any particular issue to be addressed prior to release? I don't anticipate breaking changes in the vAPI, in any case.

**Comment by jmfederico** [comment]
Can you link here the issues that would block the release? Like "unblocked connections" that you mention?

**Comment by x0wl** [comment]
But it's already in the extension gallery, am I not getting something?

**Comment by chrisgonyea** [comment]
Yeah, I see it on the extension gallery already. https://safari-extensions.apple.com/?q=ublock

**Comment by el1t** [comment]
That's not mine! I'm not sure who uploaded it. I would like to look into [ping](issues/36) and webworker requests before releasing. Been kind of busy lately, but I'll try to get it ready asap.

**Comment by jmfederico** [comment]
Is there a way to know we published it to the extension gallery?

**Comment by el1t** [comment]
@jmfederico I'll link it here and on the readme

**Comment by jmfederico** [comment]
Ah, I just noticed a mistake on my post: The question was meant to be:

Is there a way to know **who** published it to the extension gallery?

Don't know where the **we** came from.

**Comment by el1t** [comment]
@jmfederico I don't know if that's possible; the extension may have been resigned by the Extension Gallery. It's definitely an issue though, as I doubt I'm allowed to upload a duplicate extension. Hopefully whoever uploaded it will pull it...

On the other hand, you can verify the SHA-256 fingerprint of the cert that I'm using to sign the releases I upload to Github is as follows:
```
A1 E6 03 7F 11 9E CA B1 52 5D 5A B0 D9 67 05 56 95 54 0A FE 4B E5 A6 22 7B 1E D3 00 BB 34 9A 33
```

**Comment by hartator** [comment]
@el1t I am the one who published the extension.

Sorry for the trouble, I didn't understood that the extension wasn't stable enough and was surprised it wasn't already on the extension gallery. I though it was linked to you guys not having a marketing website and a proper SVG logo knowing Apple can be tricky to accept apps in their App store for these reasons. So, I made a marketing site in a few hours, re-draw a HD version of the uBlock Origin logo and submitted it to Apple. (https://hartator.github.io/uBlock-Safari/).

I had to re-sign the extension for the submission, but for what's worth I didn't modify any of the code, so it should self-update against this GitHub like the other extensions.

I don't seem to have any direct control it though, it seems they have a manual process for Safari extensions. I know about it through this thread as they also didn't sent me any email. I only know it's my submission because the HD logo they are using matches mine.

Anyway, thanks for the awesome work you guys have been doing, I am a big fan.

What course of action do you want me to take?

I can ask them to take it down by sending them an email - I am not finding a control panel like regular iOS and MacOS apps - or you can just submit here when you update: https://developer.apple.com/safari/extensions/submission/ They seem to follow a complete manual process, so a small explanation should be enough.

**Comment by el1t** [comment]
@hartator That's alright, thanks for letting me know. I would just ask that you send an email/support request to pull it, if possible. I'll try to upload it officially in a week or so. I appreciate the support!

**Comment by hartator** [comment]
Sure, I will do that tonight.

**Comment by hartator** [comment]
@el1t Ok, I've asked to remove the extension using same form as before. There is no ways to contact them directly unfortunately. Let me know if I can do anything more to help.

**Comment by DevSysEngineer** [comment]
Any update about this?

**Comment by ruchernchong** [comment]
@el1t Roughly how often will the updates be pushed to the extension gallery when there is a new release here?

**Comment by el1t** [comment]
I will try to submit an update to the extension gallery whenever I submit a release here. I have actually submitted 1.12.5b to the gallery, but it's been some time and Apple still hasn't updated. I'll post here if/when they do.

**Comment by vassudanagunta** [comment]
I just installed the uBlock Origin in the Extensions Gallery (https://safari-extensions.apple.com/?q=ublock). Note that the Gallery shows the author as "Chris Aljoudi/Raymond Hill", but links to https://github.com/el1t/uBlock-Safari. The installed version is 1.11.3.1.

Should I not use this?

**Comment by ruchernchong** [comment]
@vassudanagunta That is the version I am using. Should be fine.

**Comment by el1t** [comment]
That's a couple versions behind the latest, which you can check on the releases
page <https://github.com/el1t/uBlock-Safari/releases>. I would prefer that
you do not install the extension on the gallery for that reason, but it's
also *cannot automatically update* to any version I upload afaik because it
was signed with a different certificate. I would also add that if you are
running an older version, to please test on the latest before reporting any
issues.

**Comment by amarendra** [comment]
Hey @el1t, is this #59 intended? The issue says the update URL is linked to chrismatic.io.

**Comment by ServerWrestler** [comment]
If @gorhill or @el1t isn't involved with this on the apple site https://safari-extensions.apple.com/details/?id=com.el1t.uBlock-8782JU4WM4 then I suggested it be removed as mentioned above.  It's been a while, any luck?

**Comment by vassudanagunta** [comment]
It appears that the unauthorized publish of uBlock Origin to the Extensions Gallery by @hartator  has been pulled.  It no longer comes up on search (https://safari-extensions.apple.com/?q=ublock).

@el1t maybe you can publish the latest release now (and close this issue)?

**Comment by bryandmlee** [comment]
@vassudanagunta The direct link for the Extensions Gallery version is still "live": https://safari-extensions.apple.com/details/?id=com.el1t.uBlock-8782JU4WM4, and its install link still works though. I can second that the search query doesn't show any results now, however.

**Comment by ghost** [comment]
... also the "security" section still lists UBO. It seems to be very persistent. It was a very bad call from hartator to upload the extension without consulting el1t first it seems. I hope this can be resolved somehow as I generally would like autoupdates for the Safari fork/port of UBO.

**Comment by vassudanagunta** [comment]
Agreed. A very bad call. I created new issue #69 because it deserves priority attention.
